### ISSUE TITLE
Lint/ShadowingOuterLocalVariable: falsely triggers in if/else condition

### ISSUE THREAD

**Issue by JeanMertz** [bbatsov/rubocop#1639]
This code triggers `Lint/ShadowingOuterLocalVariable`, but there is no outer local variable `record` within the `else` scope.

```ruby
if this == true
  record = true
else
  table.each do |record|
    record.delete!
  end
end
```

**Comment by JeanMertz** [comment]
Also, the same applies to this example, which in my opinion also shouldn't trigger this cop's warning:

`var` is only set after `vars.find` finishes, so `var` inside the block is still unique at that point.

```ruby
var = vars.find { |var| var.id == 1 }
```

**Comment by jonas054** [comment]
Duplicate of #1518 and #1544.

RuboCop and Ruby are in agreement here too:

```
 /tmp$ rubocop --only Lint/ShadowingOuterLocalVariable 1639.rb 
Inspecting 1 file
W

Offenses:

1639.rb:4:18: W: Shadowing outer local variable - record.
  table.each do |record|
                 ^^^^^^
1639.rb:9:20: W: Shadowing outer local variable - var.
var = vars.find { |var| var.id == 1 }
                   ^^^

1 file inspected, 2 offenses detected
 /tmp$ ruby -wc 1639.rb 
1639.rb:4: warning: shadowing outer local variable - record
1639.rb:9: warning: shadowing outer local variable - var
Syntax OK
```

**Comment by bbatsov** [comment]
P.S. `var` is created as soon as `var =` is encountered, that's just the way Ruby works.
